python/equistore-operations/equistore/operations/divide.py

```diff
@@ -113,29 +113,43 @@ def _divide_block_block(block_1: TensorBlock, block_2: TensorBlock) -> TensorBlo
         if len(gradient_2.gradients_list()) != 0:
             raise NotImplementedError("gradients of gradients are not supported")
 
-        values_grad = []
-        for i_sample in range(len(block_1.samples)):
-            i_sample_grad_1 = _dispatch.where(
-                gradient_1.samples.column("sample") == i_sample
-            )[0]
-            i_sample_grad_2 = _dispatch.where(
-                gradient_2.samples.column("sample") == i_sample
-            )[0]
-
-            value_grad = (
-                -block_1.values[i_sample]
-                * gradient_2.values[i_sample_grad_2]
-                / block_2.values[i_sample] ** 2
+        gradient_values_1 = gradient_1.values
+        gradient_values_2 = gradient_2.values
+        if gradient_values_1.shape != gradient_values_2.shape:
```

Comment:
from the code below, it looks like you are assuming more than the same shape, you are also assuming the same values in the same order
```suggestion
        if gradient_1.samples != gradient_2.samples:
```

Reply:
This has to be assumed, no? Otherwise, in all these binary operations, what are the output samples supposed to be... those coming from gradients_1 or gradients_2? See the construction of the output TensorBlocks (that I didn't touch)

Reply:
I'm fine with making this assumption, but the code should check for the assumption it is making, and it was checking something different